### PR TITLE
Add support for adjacency matrix

### LINKED ISSUE #1553: Add support for the adjacency_matrix aggregation
When one currently tries to use the [adjacency matrix](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-adjacency-matrix-aggregation.html) aggregation, one gets this error: `elasticsearch_dsl.exceptions.UnknownDslObject: DSL class 'adjacency_matrix' does not exist in agg.`

This is not a bug, but a feature that is not yet supported by this package. It is similar to https://github.com/elastic/elasticsearch-dsl-py/issues/1512.

### PR DESCRIPTION
**Pull request by Jonakemon**
Closes #1553

### CODE REVIEW

**Comment by Jonakemon** [comment]
Fyi, I just signed the agreement.

File: tests/test_aggs.py [hunk 1]
```diff
@@ -143,17 +162,17 @@ def test_buckets_to_dict():
     a.bucket("per_author", "terms", field="author.raw")
 
     assert {
-        "terms": {"field": "tags"},
-        "aggs": {"per_author": {"terms": {"field": "author.raw"}}},
-    } == a.to_dict()
+               "terms": {"field": "tags"},
```
Comment (sethmlarson) [inline tests/test_aggs.py:165]: Please remove all the formatting that's not related to this PR?

**Review by sethmlarson — CHANGES_REQUESTED** [review]

File: tests/test_aggs.py (hunk 1 above)
Comment (Jonakemon) [reply tests/test_aggs.py]: Sorry for the inconvience! It's fixed.

**Comment by Jonakemon** [comment]
@sethmlarson is there anything you'd like me to change about this? Or is it ready to merge?

**Comment by Jonakemon** [comment]
@sethmlarson any update? Otherwise I'll close this PR.

**Comment by Jonakemon** [comment]
@sethmlarson I'm closing this PR; bummer that it couldn't be merged in a reasonable time window.

**Comment by sethmlarson** [comment]
@Jonakemon Sorry that I didn't get back to you on this, our team doesn't currently have a full-time engineer maintaining this project as our current Python maintainer is onboarding and primarily focused on the Python Elasticsearch client. Hope this experience doesn't discourage you from contributing in the future.

**Pull request closed without merging**